### PR TITLE
embeddedetcd: restructure with options/config/server pattern

### PR DESCRIPTION
**Pull request by sttts**
Towards full kcp server options/config/server pattern implementation.

### CODE REVIEW

File: pkg/embeddedetcd/config.go [hunk 1]
```diff
@@ -92,60 +80,43 @@ func (s *Server) Run(ctx context.Context, peerPort, clientPort string, listenMet
...
-		// defines the interval for etcd watch progress notify events.
-		//
-		// note:
-		// - gcp, ocp and upstream k8s set it to 5s, so we simply follow suit
-		// - in practice this value never changes so we are not exposing it as a flag/option
-		// - we enable it only when the watch cache is on otherwise it might not scale
-		//   sending an event every 5s to thousands of clients
-		cfg.ExperimentalWatchProgressNotifyInterval = 5 * time.Second
-	}
+	cfg.ForceNewCluster = o.ForceNewCluster
```
Comment (p0lyn0mial) [inline pkg/embeddedetcd/config.go:83]: what happened to `ExperimentalWatchProgressNotifyInterval` ?

Comment (p0lyn0mial) [reply pkg/embeddedetcd/config.go:83]: do you want to expose it via `EmbeddedEtcd` options?

File: pkg/server/server.go [hunk 2]
```diff
@@ -130,27 +128,19 @@ func (s *Server) Run(ctx context.Context) error {
...
-			listenMetricsURLs, err = etcdtypes.NewURLs(s.options.EmbeddedEtcd.ListenMetricsURLs)
-			if err != nil {
-				return err
-			}
+		config, err := embeddedetcd.NewConfig(s.options.EmbeddedEtcd)
+		if err != nil {
+			return err
 		}
-		embeddedClientInfo, err := es.Run(ctx, s.options.EmbeddedEtcd.PeerPort, s.options.EmbeddedEtcd.ClientPort, listenMetricsURLs, s.options.EmbeddedEtcd.WalSizeBytes, s.options.EmbeddedEtcd.QuotaBackendBytes, s.options.EmbeddedEtcd.ForceNewCluster, s.options.GenericControlPlane.Etcd.EnableWatchCache)
+		completed, err := config.Complete()
```
Comment (p0lyn0mial) [inline pkg/server/server.go:137]: okay, now the params are taken directly from the options.

Comment (sttts) [reply pkg/server/server.go:137]: today the embedded etcd does not follow the options/config/server model at all.

File: pkg/embeddedetcd/config.go (hunk 1 above)
Comment (sttts) [reply pkg/embeddedetcd/config.go:83]: fixed

**Review by p0lyn0mial — APPROVED** [review]

**Comment by sttts** [comment]
/override ci/prow/e2e-multiple-runs

Unrelated flake.

**Pull request merged**